src/app/screens/nftDetail/useNftDetail.ts

```diff
@@ -0,0 +1,111 @@
+import useStacksCollectibles from '@hooks/queries/useStacksCollectibles';
+import useNftDataSelector from '@hooks/stores/useNftDataSelector';
+import useNftDataReducer from '@hooks/stores/useNftReducer';
+import useResetUserFlow from '@hooks/useResetUserFlow';
+import useWalletSelector from '@hooks/useWalletSelector';
+import { getNftDetail, NftDetailResponse } from '@secretkeylabs/xverse-core';
+import { NftData } from '@secretkeylabs/xverse-core/types/api/stacks/assets';
+import { useMutation } from '@tanstack/react-query';
+import { GAMMA_URL } from '@utils/constants';
+import { getExplorerUrl, isLedgerAccount } from '@utils/helper';
+import { useEffect, useMemo, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+export default function useNftDetail() {
+  const navigate = useNavigate();
+  const { stxAddress, selectedAccount } = useWalletSelector();
+  const stacksNftsQuery = useStacksCollectibles();
+
+  const { id } = useParams();
+  const nftIdDetails = id!.split('::');
+  const collectionInfo = stacksNftsQuery.data?.pages
+    ?.map((page) => page?.results)
+    .flat()
+    .find((collection) => collection.collection_id === nftIdDetails[0]);
+
+  const { nftData } = useNftDataSelector();
+  const { storeNftData } = useNftDataReducer();
+  const [nft, setNft] = useState<NftData | undefined>(undefined);
+  const {
+    isLoading,
+    data: nftDetailsData,
+    mutate,
+  } = useMutation<NftDetailResponse | undefined, Error, { principal: string }>({
+    mutationFn: async ({ principal }) => {
+      const contractInfo: string[] = principal.split('.');
+      return getNftDetail(nftIdDetails[2].replace('u', ''), contractInfo[0], contractInfo[1]);
+    },
+  });
```

Comment:
I wonder why this isn't a query? usually mutations are for POST requests where we might want to show optimistic updates